File: JULY-2025/03-07-2025/p3304.rb
class String
    T = [('a'..'z').to_a, ('b'..'z').to_a + ['a']].collect {|a| a.join }
    def shift = tr(*T)
end

def kth_character(k)
    s = 'a'
    s += s.shift while s.size < k
    s[k - 1]
end

File: JULY-2025/03-07-2025/p3304.go
func kthCharacter(k int) byte {
	return 'a' + byte(bits.OnesCount(uint(k-1))%26)
}

func kthCharacter2(k int) byte {
	shift := 0
	len := bits.Len(uint(k))
	msb := 1 << (len - 1)

	for k > msb {
		k -= msb
		shift++
		len = bits.Len(uint(k))
		msb = 1 << (len - 1)
	}
	shift = (shift + len - 1) % 26
	return 'a' + byte(shift)
}